packages/expo-router/src/link/zoom/usePreventZoomTransitionDismissal.tsx

```diff
@@ -0,0 +1,41 @@
+'use client';
+
+import type { UsePreventZoomTransitionDismissalOptions } from './usePreventZoomTransitionDismissal.types';
+
+/**
+ * Limits the screen area where interactive dismissal gestures are allowed for zoom transitions.
+ *
+ * This hook must be called from the destination screen of a zoom transition (the screen you navigate to, not the source).
+ * It restricts where users can start swipe gestures to dismiss the screen and return to the previous screen.
+ *
+ * When a dismissal gesture starts inside the bounds, the screen can be dismissed. When a dismissal gesture starts outside
+ * the bounds, dismissal is blocked completely. Undefined coordinates place no restriction on that dimension.
+ *
+ * > **Note**: Only one instance of this hook should be used per screen. If multiple instances exist, the last one to render will take effect.
+ *
+ * @example
+ * ```tsx
+ * // In your destination screen (e.g., app/image.tsx)
+ * import { usePreventZoomTransitionDismissal } from 'expo-router';
+ * import { useWindowDimensions, Image } from 'react-native';
```

Comment:
```suggestion
 * import { useWindowDimensions } from 'react-native';
  * import { Image } from 'expo-image';
```

In other examples, we use `Image` from `expo-image`, so follow that here as well for consistency.